hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -485,4 +497,12 @@ public Fees calculateFees(@NonNull final FeeContext feeContext) {
                         txnEstimateFactory)
                 .usageGiven(fromPbj(body), sigValueObj, token));
     }
+
+    private boolean checkForZeroAccountIfContractCall(boolean isContractCall, TokenUpdateTransactionBody op) {
```

Comment:
Is this named right?   First, `check...` doesn't tell you by the name whether its _true_ for a zero account if contract call, or false.  Should be something that tells you definitely like `is` or `has` or `can`, or something like that. 

But also, the test seems to be "is a contract call or else it's not zero" or alternatively, "its not zero, except if it is a contract call then it can be zero".

But then OTOH the comment at L365-367 says zero is valid _unless_ it is a contract call.

Am I reading this right?   (... and I've lost track now of what it _should_ be ...)

Comment indicates name of this method could be `canBeZeroUnlessAContract`?  And then the code needs to match that.  If the comment is correct.

Reply:
I also have a question about the argument named `isContractCall`.  Because at L369 the actual argument is `isHapiCall` which I don't get why that means it's a contract call (now that I look at it again), and at L172 and L117 the actual argument is `txn.hasTransactionId()` yet the comment at L115 says that if the transaction id is _missing_ then it is a contract call, so now I'm confused.